### PR TITLE
fix some `wof:hierarchy`

### PR DESCRIPTION
**Pull request by Joxit**
These changes are based on `frgov:LIB` property and some local knowledge

### CODE REVIEW

**Comment by stepps00** [comment]
Nice, thank you @joxit!

Are you able to "exportify" these changes? Running each file in this branch against the wof-exportify tool will update the `"wof:lastmodified"` date (and validate the changes), which we need to do before import. The updated lastmodified date helps WOF track recent changes and are needed for the dist files.

https://github.com/whosonfirst/py-mapzen-whosonfirst-export/blob/master/scripts/wof-exportify

**Comment by Joxit** [comment]
Hi @stepps00 ! It's a pleasure :smile: 

Oh, thank you for this link, I was looking for it ! I did the exportify only the changed files.
I saw some `"mz:is_current":-1` and a decimal reduction.

**Comment by stepps00** [comment]
Thanks @joxit -- I'll take a look this week at merging this.

> I saw some "mz:is_current":-1 and a decimal reduction.

Both of these changes are due to the Exportify tool. The tool ensures a `mz:is_current` property is present on the record and shortens the decimal precision in some cases.

Just out of curiosity, how were the new hierarchy values calculated? WOF has the concept of a [wof:controlled](https://github.com/whosonfirst/whosonfirst-properties/blob/master/properties/wof/controlled.json) property that we could add to ensure these hierarchies do not change in the future. It may be worthwhile to look into adding this in a handful of special cases.

**Comment by Joxit** [comment]
Yeah! Ok ! :smile: 

I wanted to know if the WOF postalcodes for France were coorect.
So I wrote a little [script](https://github.com/Joxit/whosonfirst-utils/blob/master/examples/la-poste-postalcode.js) to compare WOF and our official postal code data from [La Poste](https://datanova.laposte.fr/explore/dataset/laposte_hexasmal/information/?disjunctive.code_commune_insee&disjunctive.nom_de_la_commune&disjunctive.code_postal&disjunctive.libell_d_acheminement&disjunctive.ligne_5&location=12,45.55751,0.98276&basemap=jawg.streets).
The script uses pelias structured search API.

I found errors, some of them due to abbreviations (such as St. instead of Saint) or missing quote or a rename of the locality. So I decided to check all of them by hand.
There were 726 issues, I did all these 726 entries one by one to eliminate false positives...

1. Do a group by postal code on La Poste data (many localities can share the same postal code)
2. Check the name list of locality/localadmin from La Poste and WOF
3. Compare the `frgov:LIB` properties from WOF and La Poste
4. Check if it's a new locality name...

It took a few hours :sweat_smile:

File: data/421/296/621/421296621.geojson [hunk 1]
```diff
@@ -34,35 +35,35 @@
...
-    "wof:geomhash":"579e36eda2bd0dbcd09a4aced1e6df8a",
+    "wof:geomhash":"93e6f5b0a784824c552d4b21a1559cf0",
     "wof:hierarchy":[
         {
             "continent_id":102191581,
             "country_id":85633147,
             "county_id":102067553,
             "empire_id":136253037,
-            "localadmin_id":404358937,
+            "localadmin_id":404358941,
```
Comment (stepps00) [inline data/421/296/621/421296621.geojson:45]: So in this case, we're saying 

https://spelunker.whosonfirst.org/id/421296621/

is now parented by localadmin of 

https://spelunker.whosonfirst.org/id/404358941/

This seems correct if we had a single one-to-one mapping of postalcode to localadmin, but it looks like the postalcode in this case is actually _larger_ than the localadmin, no? I'm fine with this edit, as we're adding a "more correct" hierarchy, but ideally this postalcode would have more than one parent/hierarchy. Similar to what we do with New York City:

https://spelunker.whosonfirst.org/id/85977539/

File: data/421/300/519/421300519.geojson [hunk 2]
```diff
@@ -35,35 +36,35 @@
...
-    "wof:geomhash":"fd267690b02021ca6cbfbaeb845e0d48",
+    "wof:geomhash":"f380a932d9fd515d572c6176e3b2126c",
     "wof:hierarchy":[
         {
             "continent_id":102191581,
             "country_id":85633147,
             "county_id":102067427,
             "empire_id":136253037,
-            "localadmin_id":404421605,
+            "localadmin_id":404359359,
```
Comment (stepps00) [inline data/421/300/519/421300519.geojson:46]: Similar case here. We're suggesting this postalcode

https://spelunker.whosonfirst.org/id/421300519/

is now parented by locaadmin:

https://spelunker.whosonfirst.org/id/404359359/

When in reality, it looks like most of these are in the same postalcode shape:

https://spelunker.whosonfirst.org/id/102066429/descendants/?exclude=nullisland&placetype=locality

Is that true? Again, these edits look fine.. I just want to understand if we should file a follow up to untangle multiple-hierarchies in France postalcodes.

**Comment by stepps00** [comment]
There is some follow up work we can do to improve the hierarchies even more for postalcode records in France, but for now, this looks OK to merge.

Thoughts @nvkelso ?

**Comment by nvkelso** [comment]
I'm fine merging. Thanks @Joxit!

The multiple hierarchy is more of an existential question that applies to existing data, too.

File: data/421/296/621/421296621.geojson (hunk 1 above)
Comment (Joxit) [reply data/421/296/621/421296621.geojson:45]: Oh, I did not know it! Thanks for the information, I would probably do another pass to manage the multiple hierarchy! 

Thank you for the example

File: data/421/300/519/421300519.geojson (hunk 2 above)
Comment (Joxit) [reply data/421/300/519/421300519.geojson:46]: Yes, that's totally right, a postalcode (not all postalcodes) can contains many localadmins.

So they may use multiple hierarchies. The shape of a postal code should fit perfectly the union of their localadmins. (In WOF, localadmins are more accurate than postalcodes)

**Comment by Joxit** [comment]
You are welcome :slightly_smiling_face: 

I think the next step can be the use of multiple hierarchy, this should not be complicated with the point in polygon service and La Poste data (to avoid mistakes).

**Pull request merged**